common/mb_controller.c

```diff
@@ -198,14 +213,21 @@ int mb_load_controller_config(pid_parameters_t* pid_params){
 *******************************************************************************/
 
 int mb_controller_update(mb_state_t* mb_state, mb_setpoints_t* mb_setpoints){  
-    // calculate error
-    float r_error, l_error;
-    r_error = mb_setpoints -> fwd_velocity - mb_state -> right_velocity; 
-    l_error = mb_setpoints -> fwd_velocity - mb_state -> left_velocity ; 
     
-    // march values
-    mb_state -> left_cmd = rc_filter_march(&l_wheel_speed_pid, l_error);
-    mb_state -> right_cmd = rc_filter_march(&r_wheel_speed_pid, r_error);
+    //mb_setpoints->left_velocity = (2 * mb_setpoints->fwd_velocity - WHEEL_BASE * mb_setpoints->turn_velocity) / 2;
+    //mb_setpoints->right_velocity = (2 * mb_setpoints->fwd_velocity + WHEEL_BASE * mb_setpoints->turn_velocity) / 2;
+
+    mb_setpoints->left_velocity = (2 * rc_filter_march(&lp_filt_r, mb_setpoints->fwd_velocity) - WHEEL_BASE * mb_setpoints->turn_velocity) / 2;
+    mb_setpoints->right_velocity = (2 * rc_filter_march(&lp_filt_l, mb_setpoints->fwd_velocity) + WHEEL_BASE * mb_setpoints->turn_velocity) / 2;
```

Comment:
should we split up this calculation?

Reply:
We could, but it doesn't make too much of a difference.